### PR TITLE
Add 0.2 & CoffeJam blogpost

### PR DESCRIPTION
**Pull request by roby2014**
Every member that contributed with new features should write-up something simple but descriptive of the new feature (images, usage, technical information, ...).

Stuff to talk about/showcase:
- [x] Observers, show callback-like behavior (@RiscadoA).
- [x] Repeating systems and fixed-step updates (@joaomanita).
- [x] New plugin scheme, dependency specification (@RiscadoA).
- [x] World inspector upgrades - ChildOf hierarchy and search bar (@diogomsmiranda).
- [x] New rendering plugins, optimizations made (@RiscadoA, @tomas7770).
- [x] Transform gizmo enhancements - constant scale and rotation controls (@DiogoMendonc-a).
- [x] New physics solver, show new complex sample (@fallenatlas).
- [x] Sub-scene importing in scene editor (@teres4).
- [x] Replaced all old serialization code by proper reflection (@Dacops).

### CODE REVIEW

File: content/0-2-and-coffeejam.rst [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+Finally, the transform gizmo is now always rendered at the same size, regardless of the selected entity's distance to the camera.
+This should hopefully make it easier to use the tool when moving entities either very far away, or very close.
+
+.. image:: images/transform_gizmo.gif
+
+World Inspector Overhaul :dim:`(@diogomsmiranda)`
+-------------------------------------------------
+
+The World inspector as been on our radar for quite some time now, as a tool that could be improved.
```
Comment (DiogoMendonc-a) [inline content/0-2-and-coffeejam.rst]:
```suggestion
The World inspector has been on our radar for quite some time now as a tool that could be improved.
```

File: content/0-2-and-coffeejam.rst [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+This release we particularly aimed on making it better by focusing on:
+
+* Making it easier to find the entities you are looking for.
```
Comment (DiogoMendonc-a) [inline content/0-2-and-coffeejam.rst]:
```suggestion
* Making it easier to find the entities you are looking for;
```

File: content/0-2-and-coffeejam.rst [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+* Getting more information about the entities in the scene.
+
+To tackle the first point I've added a search bar that allows you to filter the entities in the scene, either by their name, or by their components. i.e:
```
Comment (DiogoMendonc-a) [inline content/0-2-and-coffeejam.rst]:
```suggestion
To tackle the first point we've added a search bar that allows you to filter the entities in the scene, either by their name, or by their components. i.e:
```
Should we always use the plural, for consistency?

Comment (DiogoMendonc-a) [inline content/0-2-and-coffeejam.rst]:
```suggestion
To tackle the first point I've added a search bar that allows you to filter the entities in the scene, either by their name or by their components. e.g.:
```

File: content/0-2-and-coffeejam.rst [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+.. image:: images/transform_gizmo.gif
+
+World Inspector Overhaul :dim:`(@diogomsmiranda)`
+-------------------------------------------------
+
+The World inspector as been on our radar for quite some time now, as a tool that could be improved.
+This release we particularly aimed on making it better by focusing on:
+
+* Making it easier to find the entities you are looking for.
+* Getting more information about the entities in the scene.
```
Comment (DiogoMendonc-a) [inline content/0-2-and-coffeejam.rst]:
```suggestion
* Displaying more information about the entities in the scene.
```

File: content/0-2-and-coffeejam.rst [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+-----------------------------------------------------
+
+Previously our scene editor was lacking support for importing sub-scenes.
+This meant that developers had to manually open the scene file and paste the sub-scene UUID in its import section.
+To save time, I've added a popup that shows all the importable sub-scenes, and allows the developer to select the one they want to import.
+
+New 0.2 Engine Features
+=======================
+
+Our focus on this release was to continue working on improving and implementing base functionality, such as rendering and physics, and also to improve on the tools available to the developer.
```
Comment (DiogoMendonc-a) [inline content/0-2-and-coffeejam.rst]: Shouldn't this be at the top?

File: content/0-2-and-coffeejam.rst [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+We've compiled here some of the most important changes (if you're curious, you can check out the full changelog in our `repository <https://github.com/GameDevTecnico/cubos/blob/main/CHANGELOG.md>`_).
+
+Physics Improvements :dim:`(@fallenatlas)`
+------------------------------------------
+
+This update we introduced the ``Solver``, which works to satisfy the physics constraints applied to entities (eg. Spring, Fixed distance between entities, etc.).
```
Comment (DiogoMendonc-a) [inline content/0-2-and-coffeejam.rst]:
```suggestion
This update we introduced the ``Solver``, which works to satisfy the physics constraints applied to entities (e.g. Spring, Fixed distance between entities, etc.).
```

File: content/0-2-and-coffeejam.rst [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+This way the subgroup's systems will repeat a total of n x m times, where n is the number of times the parent tag repeats and m is the number of times the subgroup tag repeats.
+
+With this implemented, it was easy to implement the fixed-step plugin on the engine side. It simply adds a tag that makes systems repeat as needed according to
+the ``DeltaTime`` passed, avoiding variance due to different frame rates and more/less powerful PCs.
+
+Moving Away from the Old Serialization :dim:`(@Dacops)`
+-------------------------------------------------------
+
+`Serialization <https://gamedevtecnico.github.io/cubos/docs/group__core-data-ser.html>`_ is a crucial part of the game engine that allows for the saving of any CUBOS. game components as well as then loading them in.
```
Comment (DiogoMendonc-a) [inline content/0-2-and-coffeejam.rst]:
```suggestion
`Serialization <https://gamedevtecnico.github.io/cubos/docs/group__core-data-ser.html>`_ is a crucial part of the game engine that allows for the saving of any CUBOS game components as well as then loading them in.
```

File: content/0-2-and-coffeejam.rst [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+Previously, serialization was done through a series of functions, both for reading and writing that could be overloaded for each type.
```
Comment (DiogoMendonc-a) [inline content/0-2-and-coffeejam.rst]:
```suggestion
Previously, serialization was done through a series of functions, both for reading and writing, that could be overloaded for each type.
```

**Review by DiogoMendonc-a — CHANGES_REQUESTED** [review]
Minor grammar nitpicks, mostly

**Comment by RiscadoA** [comment]
Just missing an image/GIF for the scene editor (@teres4) and world inspector (@diogomsmiranda).
Also, @fallenatlas could you take increase the physics sample GIF size? It is a tad smaller than the other embedded stuff which looks a little weird.

**Review by DiogoMendonc-a — APPROVED** [review]

**Pull request merged**